drivers/hmis/app/graphql/mutations/delete_service_type.rb

```diff
@@ -15,8 +15,15 @@ def resolve(id:)
 
       service_type = Hmis::Hud::CustomServiceType.find(id)
 
-      # TODO: Eventually this should be a user-facing ValidationError returned in the {errors:} object
-      raise 'Cannot delete a service type that has services' if service_type.custom_services.exists?
+      # Can't delete HUD service type
+      access_denied! if service_type.hud_service?
```

Comment:
To confirm - is `access_denied` the right error to throw in these mutations? I used it bc the UI will prevent these actions, so the spirit feels correct: if the user tries to do this, they are attempting something they are not allowed to do. However, it could be confusing because it's not a permissions issue - no user has access to do this.

Reply:
I think it's usually preferable to have a descriptive error message, if only for code readability. We also get the benefit of grouping separately from "access denied" in Sentry, which we really only use for cases where the user lacks permission for the operation/object.

I suppose I can see the argument for this being a denial of "access" because we're just saying '**nobody** has access to delete HUD service types' implicitly in the code. OTOH, it doesn't really feel like an "access" check if there are NO users that can do it. Does that make sense?

In general I'd say it's great to include a descriptive message if you have something to say. In this case, for example, a descriptive error msg would remove the need for having a comment above it.

Reply:
Ok great, makes sense, thanks!